trieste/acquisition/rule.py

```diff
@@ -718,6 +718,49 @@ def state_func(
         return state_func
 
 
+class RandomSampling(AcquisitionRule[TensorType, SearchSpace, ProbabilisticModelType]):
```

Comment:
The other type error is because unlike DiscreteThompsonSampling this doesn't need to be generic on the supported model type: it supports any ProbabilisticModel. (DiscreteThompsonSampling only supports models that are supported by the thompson_sampler that's used.)

```suggestion
class RandomSampling(AcquisitionRule[TensorType, SearchSpace, ProbabilisticModel]):
```